src/Runner.Listener/MessageListener.cs

```diff
@@ -164,6 +164,27 @@ public async Task<Boolean> CreateSessionAsync(CancellationToken token)
                         }
                     }
 
+                    if (ex is TaskAgentSessionConflictException)
```

Comment:
try migration even on session conflict since there are a small number of runners (might be orchestrated by some auto-provision/scaling tools) hitting session conflict all the time.